notebooks/03-query/11-joins.ipynb

```diff
@@ -0,0 +1,2475 @@
+    "    print(f\"Error: {ex}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d6cded7c-2019-40d7-a7dd-0ebeb5d5a134",
+   "metadata": {},
+   "source": [
+    "As expected, the query fails because it cannot broadcast that much data to the workers. As the suggestion in the error indicates, the next run of the same query uses `sortMerge` for the join algorithm which does not have such limits. \n",
+    "\n",
+    "It takes a couple of minutes to complete, while it is running you can see its progress by attaching to it in the [Druid Console](http://localhost:8888/unified-console.html#workbench):\n",
+    "\n",
+    "![](assets/msq-attach.png)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "id": "9c0cc654-1c67-45e7-887f-c46f363580a7",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "          __time user_id       client_ip  prior_time_in_session  \\\n",
+      "0  1698195974755    2975  127.18.117.109          1698195968009   \n",
+      "1  1698195974783    1298  127.23.215.192          1698195971198   \n",
+      "2  1698195974867     146   127.55.111.74          1698195924646   \n",
+      "3  1698195974884    1331    127.5.135.45          1698195964931   \n",
+      "4  1698195974960      43  127.170.19.127          1698195939684   \n",
+      "5  1698195974974    3637  127.140.239.67          1698195949274   \n",
+      "6  1698195975002     990  127.167.105.54          1698195973918   \n",
+      "7  1698195975059    2570  127.203.188.95          1698195945510   \n",
+      "8  1698195975175    3857   127.61.117.97          1698195974215   \n",
+      "9  1698195975474    3767   127.62.96.252          1698195934342   \n",
+      "\n",
+      "   prior_events_in_session  \n",
+      "0                      110  \n",
+      "1                      215  \n",
+      "2                        5  \n",
+      "3                      160  \n",
+      "4                       55  \n",
+      "5                      108  \n",
+      "6                      147  \n",
+      "7                      247  \n",
+      "8                       36  \n",
+      "9                       14  \n",
+      "2023-10-26 01:09:17.090123 <<<<<<<<<<<<< \n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "datetime.datetime(2023, 10, 26, 1, 9, 17, 90123)"
+      ]
+     },
+     "execution_count": 28,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "req = sql_client.sql_request(sql)\n",
+    "req.add_context(\"sqlJoinAlgorithm\", 'sortMerge')\n",
+    "req.add_context(\"maxNumTasks\", 3)\n",
+    "result = sql_client.async_sql(req)\n",
+    "result=pandas.json_normalize(result.rows)\n",
+    "print_timestamp()\n",
+    "result"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "03a60d03-a8df-432e-85de-e9ffaf1075f2",
+   "metadata": {},
+   "source": [
+    "## UNION ALL Queries\n",
```

Comment:
Suggest this entire section be added to the end of the existing UNION ALL notebook.

/03-query/05-UnionOperations.ipynb

Reply:
I think it rounds out the subject well with, "you don't always need a join". While this example can also be in the UNION notebook, I think it fits in well here.